### PR TITLE
feat: remove collection support for oauth

### PR DESCRIPTION
**Pull request by PabloHiro**
##### ISSUE TYPE

 - Breaking Change 

##### COMPONENT NAME

 - Collection

##### AWX VERSION

```
TBD
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by AlanCoding** [comment]
From output:

```
Run command with data: /root/.ansible/test/venv/sanity.runtime-metadata/3.12/c33cecd0/bin/python /root/ansible/test/lib/ansible_test/_util/controller/sanity/code-smell/runtime-metadata.py
ERROR: Found 1 runtime-metadata issue(s) which need to be resolved:
ERROR: meta/runtime.yml:0:0: Removal version must be a semantic version (https://semver.org/) for dictionary value @ data['plugin_routing']['inventory']['token']['tombstone']['removal_version']. Got 'X.Y.Z'
```

Our collection often can not comply with the Ansible sanity rules because of the dual-versioning system where it vendors as both `ansible.controller` and `awx.awx` which have different versions.

**Review by jessicamack — APPROVED** [review]

**Comment by PabloHiro** [comment]
@AlanCoding I think I need the CLI PR merged first. The CI fails when testing the `export` module, which still considers `application` as an exportable asset.

File: awx_collection/README.md [hunk 1]
```diff
@@ -46,6 +63,7 @@ Notable releases of the `awx.awx` collection:
  - 11.0.0 has no non-deprecated modules that depend on the deprecated `tower-cli` [PyPI](https://pypi.org/project/ansible-tower-cli/).
  - 19.2.1 large renaming purged "tower" names (like options and module names), adding redirects for old names
  - 21.11.0 "tower" modules deprecated and symlinks removed.
+ - 25.0.0 "token" module has been removed as oauth is no longer supported, use basic auth instead
```
Comment (hakbailey) [inline awx_collection/README.md]: Should we also note the application module removal here?

File: awx_collection/tools/roles/template_galaxy/templates/README.md.j2 [hunk 2]
```diff
@@ -52,12 +69,14 @@ Notable releases of the `{{ collection_namespace }}.{{ collection_package }}` co
  - 11.0.0 has no non-deprecated modules that depend on the deprecated `tower-cli` [PyPI](https://pypi.org/project/ansible-tower-cli/).
  - 19.2.1 large renaming purged "tower" names (like options and module names), adding redirects for old names
  - 21.11.0 "tower" modules deprecated and symlinks removed.
+ - 25.0.0 "token" module has been removed as oauth is no longer supported, use basic auth instead
```
Comment (hakbailey) [inline awx_collection/tools/roles/template_galaxy/templates/README.md.j2]: Same question here about the application module.

File: awx_collection/plugins/module_utils/controller_api.py
Comment (hakbailey) [inline awx_collection/plugins/module_utils/controller_api.py:1]: There's a reference to `o_auth2_access_token` in line 319 that should probably be removed.

**Review by hakbailey — COMMENTED** [review]
Just a couple of small comments, otherwise this looks good.

File: awx_collection/tools/roles/template_galaxy/templates/README.md.j2 (hunk 2 above)
Comment (PabloHiro) [reply awx_collection/tools/roles/template_galaxy/templates/README.md.j2]: Updated, nice catch!

File: awx_collection/README.md (hunk 1 above)
Comment (PabloHiro) [reply awx_collection/README.md]: Updated, nice catch!

File: awx_collection/plugins/module_utils/controller_api.py
Comment (PabloHiro) [reply awx_collection/plugins/module_utils/controller_api.py:1]: Indeed, nice find :)!

**Comment by AlanCoding** [comment]
> The CI fails when testing the export module, which still considers application as an exportable asset.

That's just a single failure, and a legitimate point of overlap between the two PRs. I don't see any problem with us both making the same obvious change (deleting applications from options and use), and it likely won't require manual intervention in the rebase.

**Review by hakbailey — APPROVED** [review]

**Review by AlanCoding — CHANGES_REQUESTED** [review]
One collection test needs "application" removed in one place and then this is otherwise in good shape.

**Comment by PabloHiro** [comment]
> One collection test needs "application" removed in one place and then this is otherwise in good shape.

@AlanCoding  The parameters are retrieved automatically from an API/and-or whatever is available in AWXKIT so that is why we would need to merge https://github.com/ansible/awx/pull/15613 (or maybe https://github.com/ansible/awx/pull/15588 too) first before the CI passes

**Comment by AlanCoding** [comment]
https://github.com/ansible/awx/pull/15613 has another blocker that I'd like to not block this with. Maybe you can skip around that test or something temporarily since it'll be resolved by the other merge anyway.

Otherwise I was expecting this to merge first, but we can get together and coordinate order if needed. I'll go ahead and approve because my comment was self-evident from the checks anyway.

**Review by AlanCoding — APPROVED** [review]
Remaining issues are related to merge-order coordination. This should be otherwise ready to go and I don't see any reason to delay.

**Comment by AlanCoding** [comment]
Request to merge into feature branch

https://github.com/ansible/awx/pull/15647

File: awx_collection/plugins/module_utils/awxkit.py [hunk 3]
```diff
@@ -33,12 +33,8 @@ def __init__(self, argument_spec, **kwargs):
...
     def authenticate(self):
         try:
-            if self.oauth_token:
-                self.connection.login(None, None, token=self.oauth_token)
-                self.authenticated = True
-            elif self.username:
-                self.connection.login(username=self.username, password=self.password)
-                self.authenticated = True
+            self.connection.login(username=self.username, password=self.password)
+            self.authenticated = True
```
Comment (AlanCoding) [inline awx_collection/plugins/module_utils/awxkit.py:37]: This is the merge conflict and my branch actually has:

> # MERGE: fix conflicts with removal of OAuth2 token from collection branch

So there's no lack of clarity there (take this diff). We just need to formally have that happen.

**Pull request merged**